activerecord/test/cases/adapters/abstract_mysql_adapter/connection_test.rb

```diff
@@ -216,6 +216,51 @@ def test_release_non_existent_advisory_lock
       "expected release_advisory_lock to return false when there was no lock to release"
   end
 
+  def test_version_string
+    @connection.stub(:get_full_version, "8.0.35-0ubuntu0.22.04.1") do
+      assert_equal "8.0.35", @connection.get_database_version.to_s
+    end
+
+    @connection.stub(:get_full_version, "5.7.0") do
+      assert_equal "5.7.0", @connection.get_database_version.to_s
+    end
+  ensure
+    @connection.pool.pool_config.server_version = nil
+  end
+
+  def test_version_string_with_mariadb
+    @connection.stub(:get_full_version, "5.5.5-10.6.5-MariaDB-1:10.6.5+maria~focal") do
```

Comment:
Since I was in here adding test coverage, I expanded the coverage to cover the specific `5.5.5-` prefix case for MariaDB that was added in https://github.com/rails/rails/pull/29704 and took some digging to understand the specific edge case.